pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -304,11 +303,9 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream ads.AggregatedDiscove
 // an error is returned. For example, if a v2 cluster request was sent initially, then a v3 response was received, we will throw an error.
 // This is to ensure that when we do pushes, we are sending a consistent type, rather than flipping between v2 and v3.
 // A proper XDS client will not send mixed versions.
-func (s *DiscoveryServer) handleTypeURL(con *XdsConnection, typeURL string, requestedType *string) error {
```

Comment:
Maybe in a future cleanup we can use the newly added model.WatchedType instead of a *string.